### PR TITLE
Canonicalization of line separators

### LINKED ISSUE #65: Incorrect hash when signing an xml file with line breaks (windows)
I'll add a test case to reproduce the issue and more info later..

when the issue is resolved would be great to test in linux and osx to ensure that everything is :ok_hand:

### PR DESCRIPTION
**Pull request by cmordue**
Convert `\r\n` new lines to canonical form: `\n`.
Convert multiple new lines to a single one.

Fixes #65 & probably #43.

### CODE REVIEW

**Comment by bjrmatos** [comment]
awesome! can you update your PR to include this [change](https://github.com/auth0/xml-crypto/commit/55296d32cd523d597a403039cac7a58383485e75)?

File: lib/utils.js [hunk 1]
```diff
@@ -54,9 +54,13 @@ var escaped_one_to_xml_special_map = {
 }
  
 function normalizeXmlIncludingCR(string) {
-    return string.replace(/([\&"<>\r])/g, function(str, item) {
-        return xml_special_to_escaped_one_map[item];
-    })
+  // replace \r\n lines with \n
+  // See https://www.w3.org/TR/2000/WD-xml-c14n-20000119.html#charescaping
+  // and https://www.w3.org/TR/2001/REC-xml-c14n-20010315#Examples
+  return string.replace(/(\r\n|\n)+/g, '\n')
```
Comment (bjrmatos) [inline lib/utils.js]: i think the regExp would be enough with `.replace(/\r\n/g)`, since there is no need to replace `\n` with the same `\n`

Comment (cmordue) [inline lib/utils.js]: I want to convert multiple lines into a single line so to still do that this would be slightly simpler: `/(\r?\n)+/g`

Comment (bjrmatos) [inline lib/utils.js]: :+1:

**Comment by cmordue** [comment]
@bjrmatos do you have a real world example for a test for your proposal? Also, shouldn't that regex be `/(\r?\n)+/g`? since we wouldn't want to replace `\r` if it is on its own

**Comment by bjrmatos** [comment]
@cmordue you're right, i think the regex should be the one that you're suggesting

i don't have any real example for that case, just trusting in the auth0's fork :) but as i say i think that the regex should be `/(\r?\n)+/g`

**Comment by cmordue** [comment]
@bjrmatos I included the change you proposed and included some more integration tests.

Strangely, the change to remove `\r\n` from the signature value wasn't actually necessary (at least on node 4.2.4). The signature value would include the `\r` character and node's crypto lib would decide the signature matched with or without `\r` characters in the middle of the signature from the document. But it is more correct to remove the newlines from the signature value in the first place.

**Comment by bjrmatos** [comment]
> The signature value would include the \r character and node's crypto lib would decide the signature matched with or without \r characters in the middle of the signature from the document. But it is more correct to remove the newlines from the signature value in the first place.

cool! just let me test it at home and it'll be ready for merge

**Comment by bjrmatos** [comment]
@cmordue thnks for this!! you're awesome :zap:

**Comment by bjrmatos** [comment]
for a new version on npm, we'll wait until #95 is resolved

**Comment by yaronn** [comment]
:+1: good fix!

**Comment by marklagendijk** [comment]
@cmordue @yaronn @bjrmatos I believe this fix is incorrect. The documentation referenced is not the final version of c14n 1.0. The correct documentation is:
- https://www.w3.org/TR/xml-c14n#ProcessingModel (Attribute Nodes, Text Nodes) 
- https://www.w3.org/TR/xml-c14n#Example-Chars

**Pull request merged**